dev/tasks/r/azure.linux.yml

```
@@ -50,7 +50,12 @@ jobs:
         # we have to export this (right?) because we need it in the build env
         export ARROW_R_DEV={{ not_cran }}
         # Note that ci/scripts/r_test.sh sets NOT_CRAN=true if ARROW_R_DEV=TRUE
-        docker-compose run r
+        docker-compose run \
+          -e ARROW_DATASET={{ arrow_dataset|default("ON") }} \
+          -e ARROW_PARQUET={{ arrow_parquet|default("ON") }} \
+          -e ARROW_S3={{ arrow_s3|default("ON") }} \
```

This should work too right? That way by default we're testing the default, unset versions.

```suggestion
          -e ARROW_DATASET={{ arrow_dataset|default("") }} \
          -e ARROW_PARQUET={{ arrow_parquet|default("") }} \
          -e ARROW_S3={{ arrow_s3|default("") }} \
```

I'll try it and revert if it's no good.

---

I think so